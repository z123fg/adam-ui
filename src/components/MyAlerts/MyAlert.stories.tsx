import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import MyAlerts, { alertsize, alertstyle } from "./MyAlerts";

export default {
  title: "Example/MyAlerts",
  component: MyAlerts,
} as ComponentMeta<typeof MyAlerts>;

const Template: ComponentStory<typeof MyAlerts> = (args) => (
  <MyAlerts {...args}>My Alert</MyAlerts>
);

export const large = Template.bind({});
large.args = {
    alertStyle:alertstyle.Large
}

export const small = Template.bind({});
small.args = {
    alertStyle:alertstyle.Small
}
export const primary = Template.bind({});
primary.args = {
    alertSize:alertsize.Primary,
}
export const secondary = Template.bind({});
secondary.args = {
    alertSize:alertsize.Secondary
}
export const success = Template.bind({});
success.args = {
    alertSize:alertsize.Success
}
export const danger = Template.bind({});
danger.args = {
    alertSize:alertsize.Danger
}
export const warning = Template.bind({});
warning.args = {
    alertSize:alertsize.Warning
}
export const info = Template.bind({});
info.args = {
    alertSize:alertsize.Info
}
export const light = Template.bind({});
light.args = {
    alertSize:alertsize.Light
}
export const dark = Template.bind({});
dark.args = {
    alertSize:alertsize.Dark
}
 


