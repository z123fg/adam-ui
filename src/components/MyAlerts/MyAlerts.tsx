import react from 'react';
import genClassName from "../../util/genClassName";

export enum alertstyle {
    Large = "lg",
    Small = "sm",
  }


export enum alertsize {
    Primary = 'primary',
    Secondary = 'secondary',
    Success = 'success',
    Danger = 'danger',
    Warning = 'warning',
    Info = 'info',
    Light = 'light',
    Dark = 'dark',
}

interface IAlertsProps {
    alertSize?: alertsize;
    alertStyle?: alertstyle;
    className?: string;
    
}

const MyAlerts: React.FC<IAlertsProps> = (props) => {
    const {children, alertSize, alertStyle} = props;
    const classNames = genClassName('alert ', {
        [`alert-${alertSize}`]: true,
        [`alert-${alertStyle}`]: !!alertStyle,
    }).join(" ");
  
    return (
        <div className={classNames}>{children}</div>
    )
}


export default MyAlerts;