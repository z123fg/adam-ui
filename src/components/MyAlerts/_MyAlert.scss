.alert {
   width: 100%;
   $font-size-lg: $font-size-base * 1.25 !default;
   text-align: center;
   padding: 20px;
   margin:  20px 0px;

}

.alert-primary {
    @include alert-style(
        $white,
       $primary,    
       $primary,
       // $white,
       // $primary
      );
    
}
.alert-secondary{
    @include alert-style(
        $white,
        $gray-500,
        $gray-600,
        // $primary,
        // $white,
        // $primary
      );
}
.alert-success{
    @include alert-style(
        $white,
        $green,
        $gray-400,
        // $primary,
        // $white,
        // $primary
      );
}
.alert-danger{
    @include alert-style(
        $white,
        $red,
        $gray-400,
        // $primary,
        // $white,
        // $primary
      );
}
.alert-warning{
    @include alert-style(
        $white,
        $yellow,        
        $primary,
        $white,
        //$primary
      );
}
.alert-info{
    @include alert-style(
        $white,
        $cyan,
        $gray-400,
        $white,
        //$primary
      );
}
.alert-light{
    @include alert-style(
        $cyan,
        $gray-100,
        $gray-400,
        //$primary,
       // $white,
        //$primary
      );
}
.alert-dark{
    @include alert-style(
        $white,
        $gray-800,
        $gray-400,
        $gray-400,
        // $white,
        // $primary
      );
}
.alert-lg{
    // @include alertstyle(
    //     // $btn-padding-y-lg, 
    //     //  $btn-padding-x-lg,
    //     //  $btn-font-size-lg,
    //      );
    font-size: 50px;
}

.alert-sm{
    // @include alertstyle(
    //     // $Alert-padding-y-sm,
    //     // $Alert-padding-x-sm,
    //     // $Alert-font-size-sm,
         
        
    //      );
    font-size: 10px;


}

