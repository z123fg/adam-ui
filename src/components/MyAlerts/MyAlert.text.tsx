import React from "react";
import { render, fireEvent } from "@testing-library/react";
import MyAlerts, { alertsize, alertstyle } from "./MyAlerts";


/* 
jest   //general js testing
react testing library
enzyme 
mocha
chai

cypress e2e, end 2 end headless browser
*/

describe("test MyAlerts Component", () => {
    it("should snapshot", () =>{
        const wrapper = render(<MyAlerts>Default Button</MyAlerts>);
        const firstRender = wrapper.asFragment();
        expect(firstRender).toMatchSnapshot();
        expect(firstRender).toMatchInlineSnapshot(`
        <DocumentFragment>
          <div
            class="alert  alert-lg alert-sm"
            id="123"
          >
            Default alert
          </div>
        </DocumentFragment>
        `)
    })
  it("should render without issue", () => {
    const wrapper = render(<MyAlerts>Default Button</MyAlerts>);
    const buttonElement = wrapper.queryByText("Default Button");
    expect(buttonElement).toBeInTheDocument();
  });

  it("should render correct button based on the prop", () => {
    const mockOnClickHandler = jest.fn(); //it creates a mock up function
    const alertprops = {
      alertStyle: alertstyle.Large,
      alertSize: alertsize.Primary,
    };
    const wrapper = render(
      <MyAlerts
        {...alertprops}
        className="alert"        
      >
        alert type
      </MyAlerts>
    );
    const buttonElement = wrapper.queryByText("div type");
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement?.tagName).toBe("ALERT");
    expect(buttonElement).toHaveClass(
      "alert",
      "alert-secondary",
      "alert-lg",
      "alert-sm",
    );
    expect(mockOnClickHandler).toBeCalledTimes(0);
    buttonElement && fireEvent.click(buttonElement);
    expect(mockOnClickHandler).toBeCalledTimes(1);
  });

  it("link button should be rendered properly based on props", () => {
    const wrapper = render(
      <MyAlerts alertStyle={alertstyle.Large}>Link Alert</MyAlerts>
    );
    //const buttonElement = wrapper.queryByRole("link",{name:"Link Button"});
    const buttonElement = wrapper.queryByText("Link Button");
    expect(buttonElement).toHaveClass("alert", "alert-primary", "alert-sm");
  });

  it("link button should have be disabled", () => {
    const mockOnClickHandler = jest.fn();
    const wrapper = render(
      <MyAlerts
        alertStyle={alertstyle.Large}
         
      >
        Link Button
      </MyAlerts>
    );

    const buttonElement = wrapper.queryByText("Link Button");
    expect(mockOnClickHandler).toBeCalledTimes(0);
    buttonElement && fireEvent.click(buttonElement);
    expect(mockOnClickHandler).toBeCalledTimes(0);
    expect(buttonElement).toHaveClass("btn", "btn-link", "btn-sm", "disabled");
  });
});
