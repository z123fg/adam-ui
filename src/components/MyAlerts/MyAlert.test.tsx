import React from "react";
import { render, fireEvent } from "@testing-library/react";
import MyAlerts, { alertsize, alertstyle } from "./MyAlerts";


/* 
jest   //general js testing
react testing library
enzyme 
mocha
chai

cypress e2e, end 2 end headless browser
*/

describe("test MyAlerts Component", () => {
    it("should snapshot", () =>{
        const wrapper = render(<MyAlerts>Default alert</MyAlerts>);
        const firstRender = wrapper.asFragment();
        expect(firstRender).toMatchSnapshot();
       /*  expect(firstRender).toMatchInlineSnapshot(`
        <DocumentFragment>
          <div
            class="alert"
            >
            Default alert
          </div>
        </DocumentFragment>
        `) */
    })
  it("should render without issue", () => {
    const wrapper = render(<MyAlerts>Default alert</MyAlerts>);
    const alertelement = wrapper.queryByText("Default alert");
    expect(alertelement).toBeInTheDocument();
  });

  it("should render correct button based on the prop", () => {
    //const mockOnClickHandler = jest.fn(); //it creates a mock up function
    const alertprops = {
      alertStyle: alertstyle.Large,
      alertSize: alertsize.Secondary,
    };
    const wrapper = render(
      <MyAlerts
        {...alertprops}
        className="alert"        
      >
        alert type
      </MyAlerts>
    );
    const alertelement = wrapper.queryByText("alert type");
    expect(alertelement).toBeInTheDocument();
    expect(alertelement?.tagName).toBe("DIV");
    expect(alertelement).toHaveClass(
      "alert",
      "alert-secondary",
      "alert-lg"
    );
  /*   expect(mockOnClickHandler).toBeCalledTimes(0);
    alertelement && fireEvent.click(alertelement);
    expect(mockOnClickHandler).toBeCalledTimes(1); */
  });
/* 
  it("link button should be rendered properly based on props", () => {
    const wrapper = render(
      <MyAlerts alertStyle={alertstyle.Large}>Link Alert</MyAlerts>
    );
     const alertelement = wrapper.queryByText("Link alert");
    expect(alertelement).toHaveClass("alert", "alert-primary", "alert-sm");
  }); */

  // it("link button should have be disabled", () => {
  //   const mockOnClickHandler = jest.fn();
  //   const wrapper = render(
  //     <MyAlerts
  //       alertStyle={alertstyle.Large}
         
  //     >
  //       Link Button
  //     </MyAlerts>
  //   );

  //   const alertelement = wrapper.queryByText("Link alert");
  //   expect(mockOnClickHandler).toBeCalledTimes(0);
  //   alertelement && fireEvent.click(alertelement);
  //   expect(mockOnClickHandler).toBeCalledTimes(0);
  //   expect(alertelement).toHaveClass("alert", "alert-primary", "alert-sm");
  // });
});
