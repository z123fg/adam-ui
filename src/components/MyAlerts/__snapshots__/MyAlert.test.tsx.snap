// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MyAlerts Component should snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert "
  >
    Default alert
  </div>
</DocumentFragment>
`;
